### DISCUSSION TITLE
Getting error in filter condition

### DISCUSSION THREAD

**Discussion by manojkdn** [vectordotdev/vector#17925] [Q&A]
Hi, 

Could anyone help me with this error. I'm using a "filter" transform and getting an error for using >(greater than) operator whereas while using == or != I dont find any error.

ERROR: 

2023-07-10T11:57:40.493859Z  INFO vector::app: Loading configs. paths=["/root/.vector/config/vector.toml"]
2023-07-10T11:57:40.498064Z ERROR vector::topology: Configuration error. error=Transform "filter_out":
error[E100]: unhandled error
  ┌─ :1:1
  │
1 │ .time_diff > 20
  │ ^^^^^^^^^^^^^^^
  │ │
  │ expression can result in runtime error
  │ handle the error case to ensure runtime success
  │
  = see documentation about error handling at https://errors.vrl.dev/#handling
  = learn more about error code 100 at https://errors.vrl.dev/100
  = see language documentation at https://vrl.dev
  = try your code in the VRL REPL, learn more at https://vrl.dev/examples

**Comment by jszwedko** [comment]
Hi @manojkdn!

The error there is because Vector cannot be sure that `.time_diff` is a numeric value. You can do something like:

```
to_int!(.time_diff) > 20
```

to log an error and consider the condition evaluated to "false" if `.time_diff` is not an integer.

or

```
.time_diff > 20 ?? false
```

To default to `false` if `.time_diff` is not a numeric value (without logging an error). That syntax can also be used to default to `true` if `.time_diff` is not a numeric value

Hopefully this helps!
